### DISCUSSION TITLE
update node

### DISCUSSION THREAD

**Discussion by hasanshamsijeze** [vsg-dev/VulkanSceneGraph#1300] [Q&A] [answered]
Hi Robert i have problem to update Nodes.
i have used TileDatabase to create an earth. then i used vsg::Builder to create a box and show it top of the Berlin at lat: 52.5162603714634 and long  13.377684902745642 know i want to move this box to lat 51.5162603714634.
so how can i do this work ?
i propose to delete box nod and create it in new navigation but after viewer->compile(); code my program suddenly exit

**Comment by dov** [comment, marked as answer]
This was recently discussed in #1298 .

I learned, that the preferred way is not to delete and recreate, but instead to insert a MatrixTransform above the Builder generated box in the scenegraph. You can then update the `matrix` attribute of the `MatrixTransform` to move the box from one place to another, without the need to delete it.

I also learned that in order to delete the node, you have to put it into a queue that survives at least three frames. 

Make sure to turn on `windowTraits->debugLayer=true;` to see any vulkan errors.

**Comment by hasanshamsijeze** [comment]
Thank you, I was able to use the **MatrixTransform**, but there are two other problems. 
problem1 : when I move the box, the box itself does not rotate. For example, I put the box in Lat 0 and Long 0, the surface of the bottom of the box was towards the ground, but when it I took it to other Lt and Long, I expected that the surface of the box would still be towards the ground, but the box did not rotate and the corner of the box was towards the ground. In my time, I used **MatrixTransform** to rotate and filled the transfer matrix, but the rotation was in the direction of the X, Y and Z axes. the shape did not rotate around itself. How can I control this?
problem2: i want to use of Geographic coordinate (lat,long,alt) to position which functions are to convert it to XYZ 
problem3: i want  rotate it to left and right top or bottom 
the box is like an airplane that move around the earth
if there are a document to explain VSG please get it here thanks

**Comment by robertosfield** [comment]
The way to move locate objects on the earth ellipsoid in to use [vsg::EllipsoidModel](https://github.com/vsg-dev/VulkanSceneGraph/blob/master/include/vsg/app/EllipsoidModel.h).  The method specifically written for the task of positioning objects is:

~~~ cpp
        /// latitude and longitude in degrees, altitude in metres
        dmat4 computeLocalToWorldTransform(const dvec3& lla) const;
~~~

You can use this to set the MatrixTransform::matrix value, and can update this during the update phase of the frame loop. There are several vsgExamples that use this method to place objects on the ellipsoid so have a search for the use of this method.

**Comment by hasanshamsijeze** [comment]
thanks my problem is solved but is there a document to describe VSG methods else examples?

**Comment by robertosfield** [comment]
The VulkanSceneGraph Tuttoiral online book is not complete, but work is progress is here:

https://vsg-dev.github.io/vsg-dev.io/tutorials/
